src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelBinderProviderContext.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    /// <summary>
+    /// A context object for <see cref="IModelBinderProvider.Create(ModelBinderProviderContext)"/>.
+    /// </summary>
+    public abstract class ModelBinderProviderContext
+    {
+        /// <summary>
+        /// Creates an <see cref="IModelBinder"/> for the given <paramref name="metadata"/>.
+        /// </summary>
+        /// <param name="metadata">The <see cref="ModelMetadata"/> for the model.</param>
+        /// <returns>An <see cref="IModelBinder"/>.</returns>
+        public abstract IModelBinder CreateBinder(ModelMetadata metadata);
```

Comment:
Strange to surface a method on a context object. Could the `IModelBinderFactory` and `ModelBinderFactoryContext` be in Abstractions and included in this context?